### PR TITLE
#25: Add support for node-level memory constraints

### LINKED ISSUE #25: Add support for node-level memory constraints
This is to implement formula (8) in the arxiv paper:
Screenshot 2024-10-23 at 14 06 31

@lifflander

### PR DESCRIPTION
**Pull request by ppebay**
Resolves #25

### CODE REVIEW

**Comment by ppebay** [comment]
@pierrepebay can you please look into the CI issues? Thanks

**Comment by ppebay** [comment]
@lifflander @pierrepebay all issues fixed, PR can now be reviewed tx

File: src/ccm_milp_parse_json.py [hunk 1]
```diff
@@ -47,17 +46,15 @@ def run_batch(file_input_json_files: str):
...
 def main():
     """Main"""
 
     # Manage options
     parser = argparse.ArgumentParser(
         prog="CCM-MILP parse json",
-        description="Parse json data to a python class oused by problem solver"
-    )
-
+        description="Parse json data to a python class oused by problem solver")
```
Comment (pierrepebay) [inline src/ccm_milp_parse_json.py:57]: "oused" => "used"

File: src/ccm_milp/data.py [hunk 2]
```diff
@@ -95,31 +98,31 @@ def parse_json(self, data_files: list, rank_memory_bound : float):
                         obj_id = task.get("entity").get("id")
                         if task.get("user_defined") is None:
                             continue
-                        shared_id = task.get("user_defined").get("shared_id")
+                        shared_block_id = task.get("user_defined").get("shared_block_id")
```
Comment (lifflander) [inline src/ccm_milp/data.py:101]: This is not correct. It is `shared_id` in the json files.

**Review by lifflander — CHANGES_REQUESTED** [review]
Overall, this looks good to me. I just see an issue with the parser, it should be `shared_id`, not `shared_block_id`

File: src/ccm_milp/data.py (hunk 2 above)
Comment (ppebay) [reply src/ccm_milp/data.py:101]: The problem is that we use `shared_id` then we get an error with the data files stored in CCM-MILP (the block IDs get disregarded).

Could it be that the data files there are themselves outdated?

Comment (ppebay) [reply src/ccm_milp/data.py:101]: @lifflander after having spoken about the issue @pierrepebay I realized that the data files in `ccm-milp` were, in fact, outdated as they use the `shared_block_id` key that has been deprecated elsewhere. I am therefore copying the more recent data set from `LBAF` and subsequently reverting `data.py` from `shared_block_id` to `shared_id`.
That should resolve the issue. Tx

**Pull request merged**